packages/flutter/lib/src/painting/image_provider.dart

```diff
@@ -1191,6 +1191,23 @@ enum ResizeImagePolicy {
 /// Instructs Flutter to decode the image at the specified dimensions
 /// instead of at its native size.
 ///
+/// The [width] and [height] parameters refer to logical pixels.
+///
+/// Logical pixels are roughly the same visual size across devices.
+/// Physical pixels are the size of the actual hardware pixels on the device.
+/// The number of physical pixels per logical pixel is described by the `devicePixelRatio`.
+///
+/// For example, if you want to resize the image to half of the screen's width, you can use:
+///
+/// ```dart
+///    Image(
+///      image: ResizeImage(
+///        FileImage(File('path/to/image')),
+///        width: (MediaQuery.sizeOf(context).width / 2).toInt(), // Half of the screen's width.
```

Comment:
According the the `Linux analyze` task, you'll also need to define the `context` as a variable.

```
packages/flutter/lib/src/painting/image_provider.dart:1206:38: Undefined name 'context' (statement) (undefined_identifier)
Found 1 snippet code errors.
```